src/workerd/api/streams/readable.c++

```diff
@@ -293,6 +293,8 @@ ReadableStream::ReadableStream(
     kj::Own<ReadableStreamSource> source)
     : ioContext(ioContext),
       controller(kj::heap<ReadableStreamInternalController>(ioContext.addObject(kj::mv(source)))) {
+  eofResolverPair = kj::heap<jsg::PromiseResolverPair<void>>(
+      jsg::newPromiseAndResolver<void>(ioContext.getCurrentLock().getIsolate()));
```

Comment:
You really shouldn't need `kj::heap` here. the `jsg::PromiseResolverPair` doesn't need to be wrapped in a `kj::Own` here, and you'll likely want to gc visit its resolver and promise.

I'm also not sure exactly why this is needed on all `Readable` instances?

Reply:
> I'm also not sure exactly why this is needed on all Readable instances?

Is there a better place for it? I just followed Kenton's suggestion to expose it in ReadableStream.

Reply:
Adding it to `ReadableStream` is fine, but it likely doesn't need to be created for every instance that uses this constructor tho. Can it be made an option? Or created by a separate method call?